sharething_pwa/src/Home/HomeScreen.tsx

```diff
@@ -29,7 +32,7 @@ class HomeScreen extends React.Component {
         items: items,
         loading: false,
       });
-      this.setState({unsubscribe: listener});
+      this.setState({ unsubscribe: listener });
```

Comment:
Few things:
* no point in putting `unsubscribe` to state. Usually you put to state things that change over time (usually based on user interactions). Instance field is better place for unsubscription function: `this.unsubscribe = listener`
* I see there are multiple calls to `setState` in this method. Generally, that's not a good idea. `setState` is asynchronous, meaning that it doesn't change `this.state` right away, but will do it later, when React sees it appropriate. This can cause a lot of confusion in some cases - [https://reactjs.org/docs/faq-state.html](url)